Cleanup indexes on citus

emord opened the pull request:
https://dimagi-dev.atlassian.net/browse/IIO-596

##### SUMMARY
Checked out pghero today and made this to drop all the extra indexes it reported. It was particularly nasty between may and july. I think that may have been due to that issue where some queries were running twice on citus.

emord committed:
Remove duplicate state_id index from agg_awc_daily

emord committed:
Remove duplicate location indexes from UCR data sources

emord committed:
Drop duplicate index from citus

emord committed:
Fix aggregation output test

emord committed:
Mark migration as non-atomic

emord committed:
Mark correct migration as non atomic

snopoke commented on custom/icds_reports/models/aggregate.py:
```diff
@@ -200,7 +200,7 @@ class AwcLocation(models.Model, AggregateMixin):
     district_name = models.TextField(blank=True, null=True)
     district_site_code = models.TextField(blank=True, null=True)
-    state_id = models.TextField(db_index=True)
+    state_id = models.TextField()
```
doesn't this require a django migration?

emord replied:
I thought it would too, but when I ran makemigrations yesterday, it didn't show anything related to this. I think its because its an index on an unmanaged table so django doesn't actually keep any state on it. its just for documentation (though I didn't verify this)

snopoke commented on custom/icds_reports/migrations/0151_drop_duplicate_indexes.py:
```diff
@@ -0,0 +1,21 @@
+
+
+class Migration(migrations.Migration):
+    atomic = False
```
build is failing with this error:
```
django.db.utils.InternalError: DROP INDEX CONCURRENTLY cannot run inside a transaction block
```

Not sure why since you've marked this as non-atomic.

emord replied:
I spent the past hour or so digging in django internals/our migrations trying to figure this one out and could not figure this one out, but did see that we never use the atomic=False pattern with our custom migrator. i may just make this run with django's RunSQL pattern unless another idea comes to me

snopoke replied:
odd, I don't see how using the migrator would have an impact. It just returns a version of `RunSQL`

emord replied:
I'm not sure what's going on,but it does look like we never try to drop index concurrently in a transaction. I wonder if a lot of our reverse sql commands are actually broken and would fail in a similar way?

I took this out of a non-atomic migration and just dropped if it exists. Most of hte se tables are somewhat old so it should drop quickly, but I could run thsi outside of a deploy as well. Also looks like this PR is now failing on some other tests that other tests on master are failing on

emord committed:
Merge branch 'master' into je/cleanup-indexes

emord committed:
Switch to RunSQL

emord committed:
Drop index immediately

emord committed:
Drop new days of indexes

emord committed:
Add more duplicate indexes to be removed

emord committed:
Remove no longer necessary migration

emord commented:
As mentioned above I went ahead and ran the drop indexes outside of the deploy/migration process. I also found that pghero doesn't report all duplicate indexes when there are more than 2 duplicates (like if there are 4 of the same index, it may only report 1 or 2 of them at first). So after some iteration I added more indexes. The final list of indexes I dropped can be found in https://github.com/dimagi/commcare-hq/blob/086a56fe81b149dded8790bd9f30572cc23a4abc/custom/icds_reports/migrations/sql_templates/drop_duplicate_indexes.sql.

Because I ran this outside of deploy/migrations (due to the issues with making it non-atomic), I removed the migration from the code

snopoke commented:
Test failing on `FAIL: custom.icds.tests.test_ls_vhnd_survey:TestLSVHNDSurveyIndicator.test_multiple_locations
`. Merge with master to fix.

emord committed:
Merge branch 'master' into je/cleanup-indexes
